File: wlr/OGUR/OGUR/Text/TextHandler.cs
using System.Collections.Generic;
using System.Linq;
using Microsoft.Xna.Framework.Graphics;
using OGUR.Management;
using OGUR.Text;

namespace OGUR.Text
{
    public class TextHandler
    {
        private List<Text> m_contents = new List<Text>();

        public TextHandler()
        {
        }

        public void Add(Text textToAdd)
        {
            if(!m_contents.Contains(textToAdd))
            {
                m_contents.Add(textToAdd);    
            }
        }

        public void Clear()
        {
            m_contents.Clear();
        }

        public void Update()
        {
            for (var ii = 0; ii < m_contents.Count; ii++)
            {
                if(m_contents[ii].Update()<=0)
                {
                    m_contents.Remove(m_contents[ii]);
                    ii--;
                }
            }
        }

        public void Draw()
        {

            if (XnaManager.Renderer != null)
            {
                foreach (var component in m_contents)
                {
                    component.Draw();
                }
            }
        }
    }
}

File: code/aigilas/src/spx/text/TextHandler.java
package spx.text; import com.badlogic.gdx.graphics.Color;import java.util.ArrayList;import java.util.List;import spx.core.Point2;import spx.core.XnaManager;public class TextHandler {	private final DefaultHudText[] defaultPool = new DefaultHudText[100];	private int defaultIndex = 0;	private final List<Text> _contents = new ArrayList<Text>();	public TextHandler() {		for (int ii = 0; ii < defaultPool.length; ii++) {			defaultPool[ii] = new DefaultHudText(.2f);		}	}	public void WriteDefault(String contents, int x, int y, Point2 origin) {		defaultPool[defaultIndex].Reset(contents, x, y, origin);		Add(defaultPool[defaultIndex]);		defaultIndex = (defaultIndex + 1) % defaultPool.length;	}	public void Add(Text textToAdd) {		if (!_contents.contains(textToAdd)) {			_contents.add(textToAdd);		}	}	public void Clear() {		_contents.clear();	}	public void Update() {		for (int ii = 0; ii < _contents.size(); ii++) {			if (_contents.get(ii).Update() <= 0) {				_contents.remove(_contents.get(ii));				ii--;			}		}	}	public void Draw() {		if (XnaManager.Renderer != null) {			for (Text component : _contents) {				component.Draw();			}		}	}}